modules/src/core/ics02_client/client_consensus.rs

```diff
@@ -25,23 +25,30 @@ pub const TENDERMINT_CONSENSUS_STATE_TYPE_URL: &str =
 
 pub const MOCK_CONSENSUS_STATE_TYPE_URL: &str = "/ibc.mock.ConsensusState";
 
-pub trait ConsensusState: Clone + core::fmt::Debug + Send + Sync {
-    type Error;
-
+pub trait ConsensusState: core::fmt::Debug + Send + Sync + AsAny {
     /// Type of client associated with this consensus state (eg. Tendermint)
     fn client_type(&self) -> ClientType;
 
     /// Commitment root of the consensus state, which is used for key-value pair verification.
     fn root(&self) -> &CommitmentRoot;
 
-    /// Wrap into an `AnyConsensusState`
-    fn wrap_any(self) -> AnyConsensusState;
+    fn encode_vec(&self) -> Result<Vec<u8>, Error>;
+}
+
+pub trait AsAny: AnyTrait {
+    fn as_any(&self) -> &dyn AnyTrait;
+}
+
+impl<M: AnyTrait + ConsensusState> AsAny for M {
+    fn as_any(&self) -> &dyn AnyTrait {
+        self
+    }
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize)]
 #[serde(tag = "type")]
 pub enum AnyConsensusState {
-    Tendermint(consensus_state::ConsensusState),
+    Tendermint(TmConsensusState),
```

Comment:
I think we also want to eleminate `AnyConsensusState`

Reply:
Thanks for the review @seunlanlege! You're right, we want to eliminate these enums entirely. But this PR is phase 1 of the `Any*` enum removal. The enums themselves will be removed in the next PR which I am currently working on.